development/deploy-to-vm.sh

```
@@ -5,44 +5,52 @@ set -euo pipefail
 echo "  Commit:       ${GITHUB_SHA:-$(git rev-parse --short HEAD)}"
 echo "══════════════════════════════════════"
 
 # Step 1: Pull latest code
 echo "→ Pulling latest code..."
-ssh ${SSH_OPTS} "${REMOTE_USER}@${REMOTE_HOST}" "
+ssh "${SSH_OPTS[@]}" "${REMOTE_USER}@${REMOTE_HOST}" "
   export PATH=\$HOME/.local/bin:\$PATH
   cd ${REMOTE_DIR}
   git fetch origin
-  git reset --hard origin/main
+  git reset --hard "${DEPLOY_BRANCH}"
```

Shellcheck (0.11.0)

[warning] The surrounding quotes actually unquote this. Remove or escape them.

(SC2027)